### PR TITLE
feat(agent-eval): add Quality Gates with Metric Thresholds

### PR DESCRIPTION
**Pull request by svidiella**
### Summary
Introduces support for **Quality Gates** in `agent-eval` by allowing users to define strict performance/quality thresholds for each metric. If any metric falls below its configured threshold, the evaluation run is flagged as failed.

### Changes
1.  **Metric Threshold Configuration (`metric_definitions.json`)**:
    *   Added support for a `threshold` key (float) in metric definitions.
2.  **Threshold Evaluation Engine (`evaluator.py`)**:
    *   Implements logic to compare final metric scores against their defined thresholds.
    *   Generates a structured `threshold_results` map detailing which metrics passed or failed.
3.  **CLI Enforcement (`evaluate.py`, `run.py`)**:
    *   The CLI now parses and displays threshold results.
    *   If any quality gate is breached, the CLI exits with a non-zero exit code (crucial for CI/CD pipeline integration) and prints a prominent warning.
4.  **SDK & Schema Integration (`sdk.py`)**:
    *   Updated the SDK to return threshold status alongside the evaluation summary.
5.  **Robust Tests**:
    *   Added unit tests in `test_evaluator.py` and `test_sdk.py` to verify threshold passing, failing, and edge cases (e.g., missing thresholds).

### Integration Context
*   **Base Branch**: `main` (independent PR, can be reviewed and merged in parallel with PR 3).
*   **Zero Conflicts**: Has no overlapping files with the Ruff configuration (PR 3), ensuring a clean, independent merge.

### CODE REVIEW

File: tools/agent-eval/src/agent_eval/core/evaluator.py [hunk 1]
```diff
@@ -666,10 +666,16 @@ def save_metrics_summary(
...
+    thresholds = {}
     if metric_definitions:
         for name, info in metric_definitions.items():
             if not isinstance(info, dict):
                 continue
+            if "threshold" in info:
+                try:
+                    thresholds[name] = float(info["threshold"])
+                except (ValueError, TypeError):
+                    pass
```
Comment (lucieyxu) [inline tools/agent-eval/src/agent_eval/core/evaluator.py]: Should we log the error?

Comment (svidiella) [reply tools/agent-eval/src/agent_eval/core/evaluator.py]: added them as warnings

**Review by lucieyxu — APPROVED** [review]
